akka-stream-tests/src/test/scala/akka/stream/scaladsl/SynchronousFileSinkSpec.scala

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright (C) 2015 Typesafe Inc. <http://www.typesafe.com>
+ */
+package akka.stream.scaladsl
+
+import java.io.File
+
+import akka.stream.testkit.AkkaSpec
+import akka.stream.{ ActorFlowMaterializer, ActorFlowMaterializerSettings }
+import akka.util.ByteString
+
+import scala.collection.mutable.ListBuffer
+import scala.concurrent.Await
+import scala.concurrent.duration._
+
+class SynchronousFileSinkSpec extends AkkaSpec(
+  """ akka.actor.default-mailbox.mailbox-type = "akka.dispatch.UnboundedMailbox" """) {
+
+  val settings = ActorFlowMaterializerSettings(system).withDispatcher("akka.actor.default-dispatcher")
+  implicit val materializer = ActorFlowMaterializer(settings)
+
+  val TestLines = {
+    val b = ListBuffer[String]()
+    b.append("a" * 1000 + "\n")
+    b.append("b" * 1000 + "\n")
+    b.append("c" * 1000 + "\n")
+    b.append("d" * 1000 + "\n")
+    b.append("e" * 1000 + "\n")
+    b.append("f" * 1000 + "\n")
+    b.toList
+  }
+
+  "SynchronousFile Sink" must {
+    "write lines to a file" in targetFile { f ⇒
+      val completion = Source(TestLines)
+        .map(ByteString(_))
+        .runWith(Sink.synchronousFile(f))
+
+      val size = Await.result(completion, 3.seconds)
+      size should equal(6006)
+      checkFileContents(f, TestLines.mkString(""))
+    }
+
+    // TODO keep this mode, or default to delete old file (sounds unsafe)
+    "by default write into existing file" in targetFile { f ⇒
+      def write(lines: List[String] = TestLines) =
+        Source(lines)
+          .map(ByteString(_))
+          .runWith(Sink.synchronousFile(f))
+
+      val completion1 = write()
+      Await.result(completion1, 3.seconds)
+
+      val lastWrite = List("x" * 100)
+      val completion2 = write(lastWrite)
+      val written2 = Await.result(completion2, 3.seconds)
+
+      written2 should ===(lastWrite.flatten.length)
+      checkFileContents(f, lastWrite.mkString("") + TestLines.mkString("").drop(100))
+    }
+
+    "allow appending to file" in targetFile { f ⇒
+      def write(lines: List[String] = TestLines) =
+        Source(lines)
+          .map(ByteString(_))
+          .runWith(Sink.synchronousFile(f, append = true))
+
+      val completion1 = write()
+      val written1 = Await.result(completion1, 3.seconds)
+
+      val lastWrite = List("x" * 100)
+      val completion2 = write(lastWrite)
+      val written2 = Await.result(completion2, 3.seconds)
+
+      f.length() should ===(written1 + written2)
+      checkFileContents(f, TestLines.mkString("") + lastWrite.mkString("") + "\n")
+    }
+
+  }
+
+  private def targetFile(block: File ⇒ Unit) {
+    val targetFile = new File("synchronous-file-sink.tmp")
```

Comment:
writing tmp file to current directory, perhaps "target/synchronous-file-sink.tmp"